ts/features/bonus/cgn/screens/merchants/CgnMerchantsListByCategory.tsx

```diff
@@ -168,70 +186,78 @@ const CgnMerchantsListByCategory = () => {
           refreshControl={
             <RefreshControl
               style={{ zIndex: 1 }}
-              refreshing={
-                isLoading(onlineMerchants) || isLoading(offlineMerchants)
+              progressViewOffset={
+                Platform.OS === "ios" ? titleHeight : undefined
               }
-              onRefresh={initLoadingLists}
-            />
-          }
-        >
-          {Platform.OS === "ios" && (
-            <View
-              style={{
-                position: "absolute",
-                height: 1000,
-                backgroundColor: categorySpecs?.colors,
-                top: -1000,
-                right: 0,
-                left: 0
+              refreshing={isListRefreshing}
+              onRefresh={() => {
+                initLoadingLists();
+                setIsPullRefresh(true);
               }}
             />
-          )}
-          {categorySpecs && (
-            <View
-              onLayout={getTitleHeight}
-              style={[
-                IOStyles.horizontalContentPadding,
-                {
-                  paddingTop: insets.top,
-                  backgroundColor: categorySpecs.colors,
-                  paddingBottom: 24
-                }
-              ]}
-            >
-              <VSpacer size={48} />
-              <VSpacer size={32} />
-              <View style={[IOStyles.row, { alignItems: "center" }]}>
+          }
+          data={isListLoading && !isPullRefresh ? [] : merchantsAll}
+          keyExtractor={item => item.id}
+          ListEmptyComponent={CgnMerchantListSkeleton}
+          renderItem={renderItem}
+          ItemSeparatorComponent={() => <Divider />}
+          ListHeaderComponent={() => (
+            <>
+              {Platform.OS === "ios" && (
```

Comment:
In order to reduce the cognitive-complexity, what do you think about declaring a local component for instance `CgnMerchantListHeader` and use it here instead of declaring everything inside this arrow function?

Reply:
done